src/main/java/com/fasterxml/jackson/databind/util/BeanUtil.java

```diff
@@ -31,11 +33,17 @@ public static String okNameForGetter(AnnotatedMember am) {
 
     public static String okNameForRegularGetter(AnnotatedMember am, String name)
     {
+        if (RecordUtil.isRecord(am.getDeclaringClass()) &&
+                Arrays.asList(RecordUtil.getRecordComponents(am.getDeclaringClass())).contains(name)) {
+            // record getters are not prefixed
```

Comment:
Jesus F Christ. Why on earth does Record spec NOT follow Bean specification?!?!!? :-( :-( :-(

Reply:
I guess this is why it would pay to follow spec work & send feedback. I would have strongly preferred not changing JDK conventions in new releases for what seem like no good reason. Existing tooling would have benefited a lot from actually following up well-established pattern. Sigh.

Reply:
But as to this part, while this would work, I'll think of a better approach: this is bit too low-level a place to apply it.
Annotation introspector could work but seems suboptimal too. Maybe `POJOPropertiesCollector` needs to handle differently or something.